app/javascript/stores/session_store.ts

```diff
@@ -1,19 +1,123 @@
-import { observable, reaction } from "mobx";
-import { RootStore } from "./root_store";
+import { observable, reaction, computed, action } from 'mobx';
+import { RootStore } from './root_store';
+import { login, LoginUser, User, logout, validate } from '../api/user';
+import api from '../api/base';
+import { createBrowserHistory, Action, Location } from 'history';
+import { SynchronizedHistory, RouterStore, syncHistoryWithStore } from 'mobx-react-router';
 
+export enum LocalStorageKey {
+  User = 'db-sql-current-user',
+  Authorization = 'Authorization',
+  CryptoKey = 'Crypto-Key'
+}
 
 class SessionStore {
-  @observable currentUser: any = null;
+  @observable pCurrentUser: User | null = null;
```

Comment:
what does this `p` stand for? is the goal to not expose this user instance? if so, you could make this user reference private, i.e.:

```
@observable private user: User | null = null;
```

Reply:
yep this is indeed the better solution, thanks :)